### DISCUSSION TITLE
Homebrew install issue

### DISCUSSION THREAD

**Discussion by Billy-Night** [Homebrew/discussions#2856] [Getting started] [answered]
Hi, I am new here and new to the terminal. I have just tried to install Homebrew on my Mac Book Air m1 with Montery 12.1 and ran into some errors at the end of the install. Could someone please give me some advice as I am not massively confident with what I am doing in the terminal.
Here is the error code while Tapping homebrew/core
==> Tapping homebrew/core
remote: Enumerating objects: 1132093, done.
remote: Counting objects: 100% (100/100), done.
remote: Compressing objects: 100% (51/51), done.
error: RPC failed; curl 92 HTTP/2 stream 5 was not closed cleanly before end of the underlying stream
error: 5639 bytes of body are still expected
fetch-pack: unexpected disconnect while reading sideband packet
fatal: early EOF
fatal: fetch-pack: invalid index-pack output
Failed during: git fetch --force origin refs/heads/master:refs/remotes/origin/master

I also ran some quick checks after:
billy@billys-macbook-air ~ % command -v brew
billy@billys-macbook-air ~ % brew -v
zsh: command not found: brew
billy@billys-macbook-air ~ % brew --version
zsh: command not found: brew
billy@billys-macbook-air ~ % brew doctor
zsh: command not found: brew

So my current understanding is that it's not install (or not correctly) so would the best option be to try a reinstall or do I need to clear any data etc... Thanks

**Comment by carlocab** [comment, marked as answer]
Did you follow the `Next Steps` shown to you by the installer? It should've looked something like
```
- Add Homebrew to your PATH in ~/.zprofile:
    echo 'eval $(/opt/homebrew/bin/brew shellenv)' >> ~/.zprofile
    eval $(/opt/homebrew/bin/brew shellenv)
```

**Reply by Billy-Night** [reply]
Thanks for your response, I can't see the Next Steps or the commands you mentioned anywhere. Where would that be displayed? thanks

**Reply by carlocab** [reply]
It should be displayed when the installer finishes (with or without errors). If you didn't see it, you can just follow the steps I showed to you above. After that, try doing `brew doctor` and follow its recommendations for fixing your install (if any).

**Reply by Billy-Night** [reply]
This is the last line in the terminal from the installer : 
Failed during: git fetch --force origin refs/heads/master:refs/remotes/origin/master

Nothing after that. So you recommend just pasting that command into the terminal ? what does it doing exactly? thanks

**Reply by carlocab** [reply]
Hm, ok, that's strange. What is the output of
```
file /opt/homebrew/bin/brew
```
?

**Reply by Billy-Night** [reply]
Here's the output:

billy@billys-macbook-air ~ % file /opt/homebrew/bin/brew

/opt/homebrew/bin/brew: Bourne-Again shell script text executable, ASCII text

thanks

**Reply by Billy-Night** [reply]
I can also see from this article the response you mentioned, but there is nothing in my terminal with those instructions.
https://www.freecodecamp.org/news/install-xcode-command-line-tools/

**Reply by carlocab** [reply]
You can go ahead and do
```
echo 'eval $(/opt/homebrew/bin/brew shellenv)' >> ~/.zprofile
eval $(/opt/homebrew/bin/brew shellenv)
```
and `brew doctor` to help you fix any remaining issues.

These commands add `brew` to your `PATH` so that you can call `brew` from your terminal as just `brew` rather than the fully-qualified `/opt/homebrew/bin/brew`.

The first line (with `echo`) makes sure `brew` is added to your `PATH` every time you start your terminal, and the second line (starts with `eval`) does that for your existing session (because it wasn't added to your `PATH` when you started). You can see what commands are executed exactly with
```
/opt/homebrew/bin/brew shellenv
```
You can also skip these, but you would need to do `/opt/homebrew/bin/brew` every time you want to use `brew`, and also prepend any command that `brew` installs with `/opt/homebrew/bin` as well.

**Reply by Billy-Night** [reply]
Ok, thanks for taking the time to explain.

So I ran the command brew doctor and got the following response:

billy@billys-macbook-air ~ % brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Homebrew/homebrew-core was not tapped properly! Run:
  rm -rf "/opt/homebrew/Library/Taps/homebrew/homebrew-core"
  brew tap homebrew/core

Warning: Some taps are not on the default git origin branch and may not receive
updates. If this is a surprise to you, check out the default branch with:
  git -C $(brew --repo homebrew/core) checkout 

So I went ahead and ran the first recommendation which was:

 rm -rf "/opt/homebrew/Library/Taps/homebrew/homebrew-core"
  brew tap homebrew/core

Then ran brew doctor command again, which gave the following output:
billy@billys-macbook-air ~ % brew doctor
Error: No available tap homebrew/core.
billy@billys-macbook-air ~ %

**Reply by carlocab** [reply]
You want to follow this:
```
Warning: Homebrew/homebrew-core was not tapped properly! Run:
rm -rf "/opt/homebrew/Library/Taps/homebrew/homebrew-core"
brew tap homebrew/core
```

**Reply by Billy-Night** [reply]
Sorry, I updated my previous message to include the response to following the advised command which is : 

Error: No available tap homebrew/core.

**Reply by carlocab** [reply]
Hm. That's weird. Do
```
rm -rf /opt/homebrew/*
```
Then run the installer again.

**Reply by Billy-Night** [reply]
So I tried the recommended command and I got the response below:

```
billy@billys-macbook-air ~ % rm - rf /opt/homebrew/* 
zsh: sure you want to delete all 21 files in /opt/homebrew [yn]? y
rm: -: No such file or directory
rm: rf: No such file or directory
rm: /opt/homebrew/Caskroom: is a directory
rm: /opt/homebrew/Cellar: is a directory
rm: /opt/homebrew/Frameworks: is a directory
rm: /opt/homebrew/Library: is a directory
rm: /opt/homebrew/bin: is a directory
rm: /opt/homebrew/completions: is a directory
rm: /opt/homebrew/docs: is a directory
rm: /opt/homebrew/etc: is a directory
rm: /opt/homebrew/include: is a directory
rm: /opt/homebrew/lib: is a directory
rm: /opt/homebrew/manpages: is a directory
rm: /opt/homebrew/opt: is a directory
rm: /opt/homebrew/sbin: is a directory
rm: /opt/homebrew/share: is a directory
rm: /opt/homebrew/var: is a directory
```
Then I ran the brew doctor which gave the following:
```
billy@billys-macbook-air ~ % brew doctor
Error: No available tap homebrew/core.
Then the brew --version :
billy@billys-macbook-air ~ % brew --version
Homebrew 3.3.12-19-g51da118-dirty
Homebrew/homebrew-core N/A
```
I don't really quite understand what's going on I thought it would be uninstalled? Thanks

**Reply by SMillerDev** [reply]
You ran `rm -<space>rf` while the suggestion was `rm -rf`

**Reply by Billy-Night** [reply]
Ah yes!! thank you and sorry. 

So would I be right in saying that the rm -rf command will completely whip the selected folder from the disk?

Is there a benefit to using that command or should I use the uninstall command : 
/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/uninstall.sh)

**Reply by carlocab** [reply]
Just a note: `rm -rf` is a very destructive operation, especially when paired with globs (`*`). Please check commands carefully before running them. (I check them thrice before posting.)

**Reply by Billy-Night** [reply]
Yes understood, thank you, I've read that it could potentially whip my entire disk if used in a certain way, which would be less than ideal. 

So before I go ahead and use the rm -rf command again should I run the uninstall command? Thanks for your help

**Reply by carlocab** [reply]
You can do either.

**Reply by Billy-Night** [reply]
I ran the uninstall, which was successful but when I restarted the terminal, I received the following:

/Users/billy/.zprofile:1: no such file or directory: /opt/homebrew/bin/brew
billy@billys-macbook-air ~ % 

So, I opened the folder with the nano .zprofile and there is one line of code, which is :

eval "$(/opt/homebrew/bin/brew shellenv)"

Should I just delete this and start the re-install?

**Reply by carlocab** [reply]
You can keep it since you'll put it back anyway after reinstalling.

Just skip the next steps about adding `brew` to your `PATH`.

**Reply by Billy-Night** [reply]
Ok, all seems good.

But, I needed to add brew to my path again because there was no response from brew, so I just deleted the duplicated code after.

Everything looks good and it says ready to brew, so thanks for your help and patients.

**Reply by XUPTLiYipeng** [reply]
It totally solves the problem!Thank you very much!

**Comment by h4sh5** [comment]
disabling HTTP/2 and using HTTP/1.1 fixes the sideband packet error:

run

`git config --global http.version HTTP/1.1`

then rerun the homebrew install script.
